x-pack/plugins/ml/public/application/explorer/explorer_charts/explorer_chart_embedded_map.tsx

```diff
@@ -0,0 +1,155 @@
+import { LayerDescriptor } from '../../../../../maps/common/descriptor_types';
+import { getMLAnomaliesActualLayer, getMLAnomaliesTypicalLayer } from './map_config';
+// import { AnomalyRecordDoc } from '../../../../common/types/seriesConfig.chartData';
+
+interface Props {
+  seriesConfig?: any; // object;
+  severity: number;
+  tooltipService: any; // object;
+}
+
+export function EmbeddedMapComponent({ seriesConfig, severity, tooltipService }: Props) {
+  const [embeddable, setEmbeddable] = useState<MapEmbeddable | ErrorEmbeddable | undefined>();
+
+  const embeddableRoot: React.RefObject<HTMLDivElement> = useRef<HTMLDivElement>(null);
+  const layerList = useRef<LayerDescriptor[]>([]);
+  const {
+    services: { embeddable: embeddablePlugin, maps: mapsPlugin },
+  } = useMlKibana();
+
+  if (!embeddablePlugin) {
+    throw new Error('Embeddable start plugin not found');
+  }
+  if (!mapsPlugin) {
+    throw new Error('Maps start plugin not found');
+  }
+
+  const factory: any = embeddablePlugin.getEmbeddableFactory(MAP_SAVED_OBJECT_TYPE);
+
+  const input: MapEmbeddableInput = {
+    id: uuid.v4(),
+    attributes: { title: '' },
+    filters: [],
+    hidePanelTitles: true,
+    refreshConfig: {
+      value: 0,
+      pause: false,
+    },
+    viewMode: ViewMode.VIEW,
+    isLayerTOCOpen: false,
+    hideFilterActions: true,
+    // Zoom Lat/Lon values are set to make sure map is in center in the panel
+    // It will also omit Greenland/Antarctica etc. NOTE: Can be removed when initialLocation is set
+    mapCenter: {
+      lon: 11,
+      lat: 20,
+      zoom: 1,
+    },
+    // can use mapSettings to center map on chart data
+    mapSettings: {
+      disableInteractive: false,
+      hideToolbarOverlay: true,
+      hideLayerControl: false,
+      hideViewControl: false,
+      // Doesn't currently work with GEO_JSON. Will uncomment when https://github.com/elastic/kibana/pull/88294 is in
+      // initialLocation: INITIAL_LOCATION.AUTO_FIT_TO_BOUNDS, // this will startup based on data-extent
+      // autoFitToDataBounds: true, // this will auto-fit when there are changes to the filter and/or query
+    },
+  };
+
+  // Update the layer list  with updated geo points upon refresh
+  useEffect(() => {
+    if (
+      embeddable &&
+      !isErrorEmbeddable(embeddable) &&
+      seriesConfig.chartData &&
+      seriesConfig.chartData.length > 0
+    ) {
+      layerList.current = [
+        layerList.current[0],
+        getMLAnomaliesActualLayer(seriesConfig.chartData),
+        getMLAnomaliesTypicalLayer(seriesConfig.chartData),
+      ];
+      embeddable.setLayerList(layerList.current);
+    }
+  }, [embeddable, seriesConfig.chartData]);
+
+  useEffect(() => {
+    async function setupEmbeddable() {
+      if (!factory) {
+        throw new Error('Map embeddable not found.');
+      }
+      const embeddableObject: any = await factory.create({
+        ...input,
+        title: 'Explorer map',
```

Comment:
Is this label shown to the user? If so, it needs to be internationalized.

Reply:
It is not shown to the user.